coucharchive

```diff
@@ -278,22 +294,39 @@ def replicate_couchdb_server(source_url, target_url,
         target_url = target_url[:-1]
 
     ignore_dbs += ('_global_changes', '_metadata', '_replicator')
-    all_dbs = [db for db in list(couchdb.Server(source_url))
-               if db not in ignore_dbs]
-
-    todos = [(source_url, target_url, db, reuse_db_if_exist) for db in all_dbs]
-
-    pool = multiprocessing.Pool(processes=NUMBER_OF_WORKERS)
-    try:  # see https://stackoverflow.com/a/25791961
-        list(pool.imap_unordered(replicate_one_database, todos))
-    except Exception as e:
-        logging.error('A replication failed, stopping...')
-        pool.close()
-        pool.terminate()
-        raise
-    else:
-        pool.close()
-        pool.join()
+    dbs = [db for db in list(couchdb.Server(source_url))
+           if db not in ignore_dbs]
+
+    control = ReplicationControl(len(dbs))
+    workers = []
+
+    while len(dbs) or len(workers):
+        while len(dbs) and len(workers) < control.ideal_number_of_workers():
+            db = dbs.pop(0)
+            worker = multiprocessing.Pool(processes=1)
+            worker.result = worker.apply_async(
+                replicate_one_database, args=(control, source_url, target_url,
+                                              db, reuse_db_if_exist))
+            workers.append(worker)
+
+        control.running_replications.value = len(workers)
+        logging.debug('Currently running %d replication workers'
+                      % len(workers))
+
+        time.sleep(1)
```

Comment:
For info I did a test were a lowered this value to `0.01` but it did not changed the outcome:
![toto](https://user-images.githubusercontent.com/10905922/67884455-26474c00-fb46-11e9-9f1e-0bc770024663.png)
(source in yellow, target in bleu)